src/test/java/com/github/bibsysdev/nva/cristin/person/CristinPersonsHandlerTest.java

```diff
@@ -0,0 +1,227 @@
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertTrue(response.getBody().contains(CristinPersonsHandler.NAME_MISSING_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    public void handlerReturnsBadRequestWhenReceivingInvalidLanguageQueryParam() throws Exception {
+        InputStream input = requestWithQueryParameters(
+            Map.of(NAME_QUERY_PARAMETER, NAME_MATTIAS, LANGUAGE_KEY, LANGUAGE_INVALID));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertTrue(response.getBody().contains(LANGUAGE_INVALID_ERROR_MESSAGE));
+    }
+
+    @Test
+    public void cristinApiClientWillStillGenerateQueryPersonsUrlEvenWithoutParameters()
+        throws IOException, URISyntaxException {
+        cristinApiClientStub.generateQueryPersonsUrl(null);
+    }
+
+    @Test
+    public void readerThrowsIoExceptionWhenReadingInvalidJson() {
+        InputStream inputStream = new ByteArrayInputStream(INVALID_JSON.getBytes(StandardCharsets.UTF_8));
+        InputStreamReader reader = new InputStreamReader(inputStream);
+        Executable action = () -> CristinApiClient.fromJson(reader, CristinPerson.class);
+        assertThrows(IOException.class, action);
+    }
+
+    @BeforeEach
+    void setUp() {
+        cristinApiClientStub = new CristinApiClientStub();
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new CristinPersonsHandler(cristinApiClientStub, environment);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(TestPairProvider.class)
+    void handlerReturnsExpectedBodyWhenRequestInputIsValid(String expected) throws IOException {
+        var actual = sendDefaultQuery().getBody();
+        assertEquals(objectMapper.readTree(expected), objectMapper.readTree(actual));
+    }
+
+    @Test
+    void handlerReturnsNonEnrichedBodyWhenEnrichingFails() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doThrow(new IOException()).when(cristinApiClientStub).getPerson(any(), any());
+        handler = new CristinPersonsHandler(cristinApiClientStub, environment);
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+        var expected = getReader(NVA_PERSONS_NON_ENRICHED_JSON_FILE);
+
+        assertEquals(objectMapper.readTree(expected), objectMapper.readTree(response.getBody()));
+    }
+
+    @Test
+    void returnNvaPersonWhenCallingNvaPersonBuilderMethodWithValidCristinPerson() throws Exception {
```

Comment:
Does this test belong here? Do we just verify that an internal method is being called in this test?

Reply:
I believe this is the base case / happy path test.